src/highlight/HighlightListener.ts

```diff
@@ -54,10 +53,8 @@ export default class HighlightListener {
         // Clear previous selection
         this.store.dispatch(setSelectionAction(null));
```

Comment:
Can we avoid passing in `store` entirely by handling this in the host component?

Reply:
This line perhaps, but `HighlightListener` depends on the store state and dispatching elsewhere in the logic. If we move the handling of events to the host component, then there's not much that is done here in `HighlightListener`

Reply:
My bad, I didn't expand the lines below the diff cutoff. Feel free to disregard.